third_party/move/move-vm/runtime/src/runtime.rs

```diff
@@ -306,20 +300,19 @@ impl VMRuntime {
                 Type::MutableReference(inner_t) | Type::Reference(inner_t) => {
                     dummy_locals.store_loc(
                         idx,
-                        self.deserialize_arg(module_store, module_storage, inner_t, arg_bytes)?,
```

Comment:
nit: this would much better be done by having a method, say as/get_reference on type, returning an Option, and then doing if let Some(inner_t) = ty.as_reference() {}, reason being not matching with open ended _ pattern (e.g. if we later added a new reference type, easier to not miss it in the types?). Nit anyway, just because of the fallthrough.

Reply:
Added (btw, the whole thing was added for Sui, technically we never pass references to entry points other than `&signer`s)